src/casp/services/api/services/cell_operations_service.py

```diff
@@ -60,36 +60,53 @@ def get_cache_info(user: models.User) -> schemas.CacheInfo:
             module_cache_info=cache_info["module_cache_info"],
         )
 
-    def __get_cell_count_from_anndata(self, file: t.BinaryIO) -> int:
+    def __validate_anndata_type(self, adata: anndata.AnnData):
         """
-        Get the number of cells in the anndata file.
+        Validates that the anndata matrix is float32, which is a requirement for cellarium-ml
 
-        :param file: Anndata file to get the number of cells from.
+        :param adata: Anndata object to validate
 
-        :return: Number of cells in the anndata file.
+        :raises exceptions.InvalidAnndataFileException: If the anndata matrix is not float32
+        """
+        if adata.X.dtype != np.float32:
+            raise exceptions.InvalidAnndataFileException(adata.X.dtype)
+        if adata.obs["total_mrna_umis"].dtype != np.float32:
+            raise exceptions.InvalidAnndataFileException(adata.obs["total_mrna_umis"].dtype)
```

Comment:
This will raise an error with message: 
`f"CAS only supports anndata matrices with dtype 'float32'. The provided matrix has dtype '{dtype}'"` 
Which seems confusing as an error message as it points to the matrix, but it is actually `obs["total_mrna_umis"]` which has to be fixed.

I think it's worth defining the error without a message in the constructor and pass a message when it's raised. So that you'd raise the error like this:
```
raise exceptions.InvalidAnndataFileException(f"CAS only supports anndata matrices with dtype 'float32'. The provided matrix has dtype '{dtype}'")
```
or 
```
f"Provided matrix's column `total_mrna_umis` should be with type of `np.float32`, got {adata.obs['total_mrna_umis']} instead"
```

Reply:
Oh you're right.  I completely missed that.  I'll update it